scss/_patterns_grid.scss

```diff
@@ -87,6 +102,12 @@
     }
   }
 
+  %span-full-grid-on-tablet-new {
+    @media (min-width: $threshold-4-small-4-med-col) {
+      grid-column: auto / span $grid-columns-medium-new;
```

Comment:
@bartaz @pastelcyborg @lyubomir-popov  This behavior here (and in another handful of "spanning placeholders") makes it hard to use `.col` instead of `.grid-col` - the `.col` class needs to know how many columns each breakpoint has available, so that the full grid can be spanned.

I suppose we could modify this so that `.row` children that extend the spanning placeholders span 4/6/12, and `.grid-row` children span 4/4/12. 

We could also keep the existing `.grid-col` classes. 

Any thoughts?

Reply:
not sure I understand - this ensures elements with no explicitly defined col-* classes occupy the entire available space. as long as the variable  on line 107 specifies the correct number of columns available, an arbitrary div placed there should fill the available space IIRC. So how does this get in the way of using say `.col-medium-3`?

Reply:
Here's a larger code snippet that shows placeholder definition and usage, from current `main`

https://github.com/canonical/vanilla-framework/blob/b90c2c651d5d1068d055fdde2e8a72d4f71a991a/scss/_patterns_grid.scss#L84-L94

This is setting `col-small-x` to span full grid width on the medium breakpoint, if no medium grid classes are used. So the placeholder must know how many columns to span, depending on the grid.

Now that I think about it more, we could maybe use `grid-column: 1 / -1` to automatically span all columns, without needing to know how many columns to span. I'll try that out

Reply:
`1 / -1` and `auto / -1` ended up not working as negative grid positions are [only supported in explicit grids](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-column#integer_custom-ident). 

To work around this, I first tried to use parent-child selectors, like this:
```scss
  %span-full-grid-on-tablet {
    @media (min-width: $threshold-4-small-4-med-col) {
      [class^='#{$grid-row-class}'] > & {
        grid-column: auto / span $grid-columns-medium;
      }
      [class^='#{$grid-row-4-cols-class}'] > & {
        grid-column: auto / span $grid-columns-medium-new;
      }
    }
  }
```

However, this increase in selector specificity necessitates changes to `vf-grid-column`:
```scss
@mixin vf-grid-column($col) {
  @supports (display: grid) {
    [class^='#{$grid-row-class}'] > & {
      grid-column-end: span #{$col};

      //nesting
      @if $col > 1 {
        & .#{$grid-row-class},
        & .#{$grid-row-4-cols-class} {
          grid-template-columns: repeat($col, minmax(0, 1fr));
        }
      }
    }
  }
}
```
All these increases in specificity added up, causing tests to fail due to large stylesheet size (failing the Parker tests). 

I've implemented another workaround: setting a CSS attribute `--vf-grid-columns` that spanning placeholders can then use to span full grid width without increasing selector specificity or increasing stylesheet size too much:

```scss
@mixin vf-b-grid-template-impl($breakpoint, $columns) {
  grid-gap: 0 map-get($grid-gutter-widths, $breakpoint);
  grid-template-columns: repeat($columns, minmax(0, 1fr));
  // Expose the number of columns available for child elements as a CSS var, so that columns can span full grid width without increasing selector specificity\
  --vf-grid-columns: #{$columns};
  // .....
}
```
```scss
  %span-full-grid-on-tablet {
    @media (min-width: $threshold-4-small-4-med-col) {
      grid-column: auto / span var(--vf-grid-columns);
    }
  }
```